### PR TITLE
libxml++: add 3.0.0

### PR DESCRIPTION
**Pull request by loskutov**
###### Motivation for this change

Version 3.0.0 of libxml++ has released and it's not yet available in nixpkgs. As it's a major release and it doesn't provide backwards compatibility with 2.\* releases, it's submitted as a new derivation.
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/libxmlxx/v3.nix
Comment (Mic92) [inline pkgs/development/libraries/libxmlxx/v3.nix]: Do you want to maintain this? `platforms` attribute is missing

Comment (zimbatm) [inline pkgs/development/libraries/libxmlxx/v3.nix]: We should avoid adding new packages with zero maintainers. Maybe @Phreedom and @wkennington are also willing to take on maintainership of that package.

**Review by kevincox — APPROVED** [review]

**Review by kevincox — COMMENTED** [review]
LGTM After @Mic92's comments and handled.

**Pull request merged**
